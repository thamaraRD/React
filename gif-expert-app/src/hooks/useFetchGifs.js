import { useEffect, useState } from "react";
import { getGifApi } from "../helpers/getGifApi";

export const useFetchGifs = ( category ) => {

  const [images, setImages] = useState([]);
  const [loader, setLoader] = useState(true);
    
  const getImages = async () => {
      const getImages = await getGifApi(category);
      setImages(getImages);
      setTimeout(() => {
        setLoader(false);
      }, 0.5);
  };

  useEffect(() => {
    
    getImages();
  // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);
  
  return {
    images,
    loader
  }


};
