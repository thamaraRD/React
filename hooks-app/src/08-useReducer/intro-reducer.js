
const initialState = [{
    id: 1,
    todo: 'aprender React',
    done: false,
}];

const todoReducer = ( state = initialState, action = {} ) => {

    if(action.type === '[TODO] add todo'){
        return [...state, action.payload];
    }


    return state;
};

let todos = todoReducer();

const newTodo = {
    id: 2,
    todo: 'mejorar en React',
    done: false
};

const addtodoAction = {
    type: '[TODO] add todo',
    payload: newTodo,
};

todos = todoReducer( todos, addtodoAction );

console.log(todos);