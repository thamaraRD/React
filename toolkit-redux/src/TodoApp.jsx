

export const TodoApp = () => {
  return (
    <>
    <h1>RTK Query- Todos</h1>
    <hr />
    <h4>isLoading...</h4>
    <pre>..</pre>
    <button>
      Next Todo
    </button>
    </>
  )
}
