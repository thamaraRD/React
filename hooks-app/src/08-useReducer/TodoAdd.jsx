import { useForm } from '../hooks/useForm';

export const TodoAdd = ({ onNewTodo }) => {

 const { description ,onInputChange, onResetForm } = useForm({
    description: '',
 })

 const onfFormSubmit = ( event ) => {
    event.preventDefault();
    if ( description.length <= 1 ) return;

    const newTodo = {
        id: new Date().getTime(),
        done: false,
        description: description,
    }

    onNewTodo(newTodo);
    onResetForm();
 }

  return (
    <form onSubmit={ onfFormSubmit }>
      <input
        type="text"
        placeholder="Â¿Que tarea por hacer?"
        className="form-control"
        id='description'
        value={description}
        onChange={ onInputChange }
      />
      <button type="submit" className="btn btn-primary mt-1">
        Agregar
      </button>
    </form>
  );
};
