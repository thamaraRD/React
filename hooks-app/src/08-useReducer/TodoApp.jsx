import { useTodo } from "../hooks/useTodo";
import { TodoAdd } from "./TodoAdd";
import { TodoList } from "./TodoList";


export const TodoApp = () => {
  
  const { todos, countAllTodo, pendingTodosCount, handleNewTodo, handleDeleteTodo, handleToggleTodo } = useTodo();

  return (
    <div>
        <h1>todo App: { countAllTodo } - <small>pendientes: { pendingTodosCount }</small></h1>
        <hr />

      <div className="row">
        <div className="col-7">
        <TodoList
        todos={ todos }
        onDeleteTodo={ handleDeleteTodo }
        onToggleTodo={ handleToggleTodo }
        />
        </div>
      
      <div className="col-5">
        <h4>Agregar TODO</h4>
        <hr />
        <TodoAdd onNewTodo={ handleNewTodo } />
      </div>
    </div>
    </div>
  )
  }
