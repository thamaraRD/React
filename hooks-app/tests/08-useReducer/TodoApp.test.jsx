import { render, screen } from "@testing-library/react"
import { TodoApp } from "../../src/08-useReducer/TodoApp"
import { useTodo } from "../../src/hooks/useTodo";

jest.mock('../../src/hooks/useTodo');

describe('testing al componente <todoApp />', () => { 
  
  useTodo.mockReturnValue({
    todos: [
      {id: 1, description: 'Todo num1', done: false},
      {id: 2, description: 'Todo num2', done: true}
    ],
    countAllTodo: 2,
    pendingTodosCount: 1,
    handleNewTodo: jest.fn(),
    handleDeleteTodo: jest.fn(),
    handleToggleTodo: jest.fn()
  })

  test('should match to snapshot', () => { 

    const { container } = render( <TodoApp /> );
    expect(container).toMatchSnapshot();

  });

  test('should de mostrar el componente correctamente', () => { 

    render( <TodoApp /> );
    // screen.debug()
    expect(screen.getByText('Todo num1')).toBeTruthy();
    expect(screen.getByText('Todo num2')).toBeTruthy();

   })


})