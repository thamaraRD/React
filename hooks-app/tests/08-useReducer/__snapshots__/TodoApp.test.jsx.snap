// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing al componente <todoApp /> should match to snapshot 1`] = `
<div>
  <div>
    <h1>
      todo App: 
      2
       - 
      <small>
        pendientes: 
        1
      </small>
    </h1>
    <hr />
    <div
      class="row"
    >
      <div
        class="col-7"
      >
        <div>
          <ul
            class="list-group"
          >
            <li
              class="list-group-item d-flex justify-content-between"
            >
              <span
                aria-label="span"
                class="align-self-center "
              >
                Todo num1
              </span>
              <button
                class="btn btn-danger"
              >
                Borrar
              </button>
            </li>
            <li
              class="list-group-item d-flex justify-content-between"
            >
              <span
                aria-label="span"
                class="align-self-center text-decoration-line-through"
              >
                Todo num2
              </span>
              <button
                class="btn btn-danger"
              >
                Borrar
              </button>
            </li>
          </ul>
        </div>
      </div>
      <div
        class="col-5"
      >
        <h4>
          Agregar TODO
        </h4>
        <hr />
        <form>
          <input
            class="form-control"
            id="description"
            placeholder="Â¿Que tarea por hacer?"
            type="text"
            value=""
          />
          <button
            class="btn btn-primary mt-1"
            type="submit"
          >
            Agregar
          </button>
        </form>
      </div>
    </div>
  </div>
</div>
`;
